text/0000-new-maintainer-group-process.md

```diff
@@ -0,0 +1,22 @@
+# Codify the Creation of New Maintainer Groups
+
+## Summary
+
+This RFC is intended to define processes for the creation of new maintainer groups.
+
+## Motivation
+
+Currently the creation of new maintainer groups is an ad hoc process that has to be hashed out every time a new group is requested. This means that intervention from the Steering Committee is often needed in situations where they should not need to weigh in. It also means that there is no standard record for the introduction of new groups meaning that the reasoning behind the formation of these new maintainer groups is lost. Codifying this process should make it easier to identify when new maintainer groups should be formed and make the process of doing so much easier. It should also give approval power to the stakeholders that will be directly affected by the introduction of new maintainer groups, namely the existing maintainer groups as opposed to the Steering Committee.
+
+## Implementation
+
+The following are the 3 scenarios in which a new maintainer group would be formed and the procedures for forming each group type.
+
+### New Maintainer Group for New Repositories
+When proposing a new project that the proposer feels does not fall under the domain of an existing maintainer group they can suggest creating a new maintainer group. As part of the proposal, members of the initial maintainer group should be suggested or a request for volunteers should be open by the proposer of the new project. This group will then be approved by the Steering Committee to be the initial set of maintainers.
+
+### New Maintainer Group for Existing Repositories that are Maintained by One Existing Group
+When the repositories exist and are already under the control of an existing maintainer team but there is a desire to have a new maintainer group take over or co-maintain a portion of the existing repositories. The process will involve opening a proposal to create a new maintainer group and outline the motivation for creating a new maintainer group and which repositories this new maintainer group would become responsible for. As part of the proposal, members of the initial maintainer group should be suggested or a request for volunteers should be open. Then the current maintainers will have to vote on whether or not they want to either cede/share control over the repositories proposed and whether or not the maintainers proposed are suitable.
```

Comment:
> Also, I feel very strongly that any group decisions should be also approved by the steering committee. The committee is responsible for guiding the project and part of that is the organization of groups. The steering committee needs to review the request and make sure it fits within the context of the entire project, not just want the specific group wants to do.

I am not deeply opposed to having the Steering Committee have approval on these proposals. I was trying to keep the Steering Committee hands off in a situation that I felt could be up to maintainer groups to decide. I am just a little curious of how you think the independent introduction of a maintainer group could drastically shape the direction of the project.

Reply:
My rationale is this:

1. Having a SC review, means we can help to ensure consistency across the project and sanity check requests. We have a view of the entire project, and not every maintainer has the same perspective.

2. I think it would be good to have a sanity check on the list of proposed members as well. I'm not sure we'd care about the individual members, but we can ensure that the introduction of a new group meets standards and requirements like having enough members to be sustainable.

3. I think the review of these should be very minimal demand on the SC. You're going to read the proposal. It's going to meet the requirements and we 👍 or it's not and we'll reject with feedback.

Reply:
My rational for approving the RFC as is was, that the group making this decision would already have the right to distribute the access permissions by doing a local vote. So I thought this can really just be left to the maintainer groups that are affected.

But if we keep SC involvement low, I am fine with that as well.

Reply:
Yes, I can definitely appreciate keeping our involvement low. I think it's worth us putting in a small bit of time here though.  We can make sure that someone is evaluating decisions and considering the big picture, it'll help with overall project consistency.

Think of the TEAMS.md doc in the community repo. It's already pretty big. If teams can split and make more teams that *could* make working with and contributing to the project more confusing. We should evaluate team decisions critically & I think that's what the SC can bring to the picture.